sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -911,9 +911,12 @@ def test_minibatch_dictionary_learning_dtype_match(
         batch_size=10,
         fit_algorithm=fit_algorithm,
         transform_algorithm=transform_algorithm,
+        max_iter=100,
```

Comment:
set max_iter to get the new behavior (old behavior doesn't have stopping criterion besides n_iter).